test/Microsoft.AspNet.Mvc.DataAnnotations.Test/DataAnnotationsModelValidatorProviderTest.cs

```diff
@@ -39,6 +39,29 @@ public void GetValidators_ReturnsValidatorForIValidatableObject()
 #endif
 
         [Fact]
+        public void GetValidators_InsertsRequiredValidatorsFirst()
+        {
+            var provider = new DataAnnotationsModelValidatorProvider(
+                new TestOptionsManager<MvcDataAnnotationsLocalizationOptions>(),
+                stringLocalizerFactory: null);
+            var metadata = _metadataProvider.GetMetadataForProperty(
+                typeof(ClassWithProperty),
+                "PropertyWithMultipleValidationAttributes");
+
+            var providerContext = new ModelValidatorProviderContext(metadata);
+
+            // Act
+            provider.GetValidators(providerContext);
+
+            // Assert
+            Assert.Equal(4, providerContext.Validators.Count);
```

Comment:
Updated this test to check if it is assignable from type `RequiredAttribute`. No need to check for specific types as we do not care about the exact order. Just need to verify if the required attributes show up first.